src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -32,35 +31,48 @@ class SubscriberArgs(parser: ArgParser) {
 
 private val PROJECT_ID = ServiceOptions.getDefaultProjectId()
 
-private val messages = LinkedBlockingDeque<PubsubMessage>()
-
-internal class MessageReceiverExample : MessageReceiver {
+internal class MessageReceiverExample(private val callback: (message: String, md5sum: String) -> Unit) : MessageReceiver {
+    private val dockerProcessor = DockerProcessor()
 
     override fun receiveMessage(message: PubsubMessage, consumer: AckReplyConsumer) {
-        messages.offer(message)
+        processMessage(message)
         consumer.ack()
     }
+
+    fun processMessage(message: PubsubMessage) {
+        val messageContent = message.data.toStringUtf8()
+        val md5sum = this.dockerProcessor.getMd5Sum(messageContent)
+
+        this.callback(messageContent, md5sum)
+    }
+}
+
+class SubscribeManager(subscriptionId: String, callback: (message: String, md5sum: String) -> Unit) {
+    private val subscriptionName = SubscriptionName.of(PROJECT_ID, subscriptionId)
+    private val receiver = MessageReceiverExample(callback)
+
+    fun subscribe() {
+        val subscriber = Subscriber.newBuilder(this.subscriptionName, this.receiver).build()
+        try {
+            subscriber.startAsync().awaitRunning()
```

Comment:
Okay, now we only need to wait until the process shuts down (currently it just exits immediately). This should do the trick (but please test if it really does)

```
  var onShutdown = CompletableFuture<Any>()
  Runtime.getRuntime().addShutdownHook(Thread(Runnable {
    onShutdown.complete(null)
  }))
 // start your subscriber
  onShutdown.get()
```

Reply:
Seems, that's working. It's not exists after start and I was able to get published message.